src/ffi.rs

```diff
@@ -142,22 +150,379 @@ extern "C" {
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn get_WKB_byte_order(&self) -> ByteOrder {
+        ByteOrder::from(unsafe { GEOS_getWKBByteOrder_r(self.ptr) })
+    }
+
+    /// Sets WKB byte order.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, ByteOrder};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    ///
+    /// context_handle.set_WKB_byte_order(ByteOrder::LittleEndian);
+    /// assert!(context_handle.get_WKB_byte_order() == ByteOrder::LittleEndian);
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn set_WKB_byte_order(&self, byte_order: ByteOrder) -> ByteOrder {
+        ByteOrder::from(unsafe { GEOS_setWKBByteOrder_r(self.ptr, byte_order.into()) })
+    }
+
+    /// Convert [`GGeom`] from WKB format.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, GGeom};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    /// let point_geom = GGeom::new("POINT (2.5 2.5)").expect("Invalid geometry");
+    /// let wkb_buf = context_handle.geom_to_WKB_buf(&point_geom)
+    ///                             .expect("conversion to WKB failed");
+    /// let new_geom = context_handle.geom_from_WKB_buf(wkb_buf.as_ref())
+    ///                              .expect("conversion to WKB failed");
+    /// assert!(point_geom.equals(&new_geom) == Ok(true));
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn geom_from_WKB_buf(&self, wkb: &[u8]) -> GeosResult<GGeom> {
+        unsafe {
+            GGeom::new_from_raw(GEOSGeomFromWKB_buf_r(self.ptr, wkb.as_ptr(), wkb.len()))
+        }
+    }
+
+    /// Convert [`GGeom`] to WKB format.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, GGeom};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    /// let point_geom = GGeom::new("POINT (2.5 2.5)").expect("Invalid geometry");
+    /// let wkb_buf = context_handle.geom_to_WKB_buf(&point_geom)
+    ///                             .expect("conversion to WKB failed");
+    /// let new_geom = context_handle.geom_from_WKB_buf(wkb_buf.as_ref())
+    ///                              .expect("conversion to WKB failed");
+    /// assert!(point_geom.equals(&new_geom) == Ok(true));
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn geom_to_WKB_buf(&self, g: &GGeom) -> Option<CVec<u8>> {
+        let mut size = 0;
+        unsafe {
+            let ptr = GEOSGeomToWKB_buf_r(self.ptr, g.as_raw(), &mut size);
+            if ptr.is_null() || size == 0 {
+                None
```

Comment:
If ptr is not null but size=0, you will leak, no?

Reply:
I'm not sure such a case is actually possible...

Reply:
as `free` a null ptr is valid, why not just returning a `CVec<u8>` and don't care of the empty case?

Reply:
Fine by me.